pyproject.toml

```diff
@@ -1,60 +1,192 @@
+
+    # https://learn.scientific-python.org/development/guides/pytest/#PP301
+    "PP301", # Has pytest in pyproject
+
+    # https://learn.scientific-python.org/development/guides/gha-basic/#GH212
+    "GH212",  # Require GHA update grouping
+
+    # https://learn.scientific-python.org/development/guides/style/#PC170
+    "PC170",  # Uses PyGrep hooks (only needed if rST present)
+
+    # https://learn.scientific-python.org/development/guides/style/#PC901
+    "PC901",  # Custom pre-commit CI message
+]
+
+
+[tool.mypy]
+strict = false
+ignore_missing_imports = true
+warn_unused_configs = true
+warn_unreachable = true
+enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+exclude = [
+
+]
+
+[tool.numpydoc_validation]
+checks = [
+    "all",  # Enable all numpydoc validation rules, apart from the following:
+
+    # -> Docstring text (summary) should start in the line immediately
+    #    after the opening quotes (not in the same line, or leaving a
+    #    blank line in between)
+    "GL01",  # Permit summary line on same line as docstring opening quotes.
+
+    # -> Closing quotes should be placed in the line after the last text
+    #    in the docstring (do not close the quotes in the same line as
+    #    the text, or leave a blank line between the last text and the
+    #    quotes)
+    "GL02",  # Permit a blank line before docstring closing quotes.
+
+    # -> Double line break found; please use only one blank line to
+    #    separate sections or paragraphs, and do not leave blank lines
+    #    at the end of docstrings
+    "GL03",  # Ignoring.
+
+    # -> See Also section not found
+    "SA01",  # Not all docstrings require a "See Also" section.
+
+    # -> No extended summary found
+    "ES01",  # Not all docstrings require an "Extended Summary" section.
+
+    # -> No examples section found
+    "EX01",  # Not all docstrings require an "Examples" section.
+
+    # -> No Yields section found
+    "YD01",  # Not all docstrings require a "Yields" section.
+
+    # TODO: exceptions that still need investigating are below. Might be fixable, or might become permanent (above):
+
+]
+exclude = [
+    '\.__eq__$',
+    '\.__ne__$',
+    '\.__repr__$',
+]
+
+[tool.ruff]
+line-length = 88
+
+[tool.ruff.format]
+preview = false
+
+[tool.ruff.lint]
+ignore = [
+    # flake8-commas (COM)
+    # https://docs.astral.sh/ruff/rules/#flake8-commas-com
+    "COM812",  # Trailing comma missing.
+    "COM819",  # Trailing comma prohibited.
+
+    # flake8-implicit-str-concat (ISC)
+    # https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/
+    # NOTE: This rule may cause conflicts when used with "ruff format".
+    "ISC001",  # Implicitly concatenate string literals on one line.
+
+
+    # TODO: exceptions that still need investigating are below. Might be fixable, or might become permanent (above):
+    "ERA001", # Has commented out code
+    "EXE001", # Shebang is present but file is not executable
```

Comment:
If this is limited to setup.py (as it was in iris-esmf-regrid), it may be worth adding to a per-file-ignores.

Reply:
yep, that's a good call. I've also added ERA001 to conf.py, as the commenting out is intentional there (for documentary purposes)

Reply:
https://github.com/SciTools/tephi/pull/172/commits/dd417efbb5d163771e07dde9694030e11f0c8d0e